lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,6 +155,18 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
+    if not (isinstance(encoded_text, list)
+            and not all(isinstance(char, int) for char in encoded_text)
+            and not isinstance(vocabulary, dict)
+            and not (isinstance(end_of_word_token, str) or end_of_word_token is None)):
+        return None
+    decoded_list = []
+    for num in encoded_text:
+        for key, value in vocabulary.items():
+            if num == value:
+                decoded_list.append(key)
+    decoded_text = ''.join(decoded_list)
+    return decoded_text.strip()
```

Comment:
what about replacing EoW token?

Reply:
thanks for reminding, forgot about this step, solving